pkg/controller/autoops/expected.go

```diff
@@ -0,0 +1,285 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package autoops
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"hash/fnv"
+	"path"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+
+	autoopsv1alpha1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/autoops/v1alpha1"
+	commonv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/v3/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/container"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/defaults"
+	common_deployment "github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/deployment"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/metadata"
+	common_name "github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/name"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/version"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/common/volume"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/controller/elasticsearch/services"
+	"github.com/elastic/cloud-on-k8s/v3/pkg/utils/k8s"
+)
+
+const (
+	autoOpsLabelName         = "autoops.k8s.elastic.co/name"
+	configVolumeName         = "config-volume"
+	configVolumePath         = "/mnt/config"
+	configHashAnnotationName = "autoops.k8s.elastic.co/config-hash"
+	readinessProbePort       = 13133
+)
+
+var (
+	// ESNAutoOpsNamer is a Namer that generates names for AutoOps deployments
+	// according to the Policy name, and associated Elasticsearch name.
+	AutoOpsNamer = common_name.NewNamer("autoops")
+	// Default resources for the AutoOps Agent deployment.
+	// These currently mirror the defaults for the Elastic Agent deployment.
+	defaultResources = corev1.ResourceRequirements{
+		Limits: map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceMemory: resource.MustParse("400Mi"),
+			corev1.ResourceCPU:    resource.MustParse("200m"),
+		},
+		Requests: map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceMemory: resource.MustParse("400Mi"),
+			corev1.ResourceCPU:    resource.MustParse("200m"),
+		},
+	}
+)
+
+func (r *ReconcileAutoOpsAgentPolicy) deploymentParams(ctx context.Context, policy autoopsv1alpha1.AutoOpsAgentPolicy, es esv1.Elasticsearch) (appsv1.Deployment, error) {
+	v, err := version.Parse(policy.Spec.Version)
+	if err != nil {
+		return appsv1.Deployment{}, err
+	}
+	labels := map[string]string{
+		commonv1.TypeLabelName: "autoops-agent",
+		autoOpsLabelName:       policy.Name,
+	}
+
+	// Create ES-specific config map volume
+	configMapName := fmt.Sprintf("%s-%s-%s", autoOpsESConfigMapName, es.Namespace, es.Name)
```

Comment:
Use namer. If users have long namespace or Elasticsearch cluster names, this could exceed Kubernetes' 253 character limit for DNS subdomain names.